Source/Core/Core/HW/SystemTimers.cpp

```diff
@@ -109,10 +109,26 @@ int GetAudioDMACallbackPeriod()
          (Mixer::FIXED_SAMPLE_RATE_DIVIDEND * 4 / 32);
 }
 
+static TimePoint next_audio_update(DT::zero());
+
 void AudioDMACallback(Core::System& system, u64 userdata, s64 cyclesLate)
 {
-  DSP::UpdateAudioDMA();  // Push audio to speakers.
-  system.GetCoreTiming().ScheduleEvent(GetAudioDMACallbackPeriod() - cyclesLate, et_AudioDMA);
+  const TimePoint now = Clock::now();
+
+  if (now > next_audio_update)
```

Comment:
I'd look at the fifo state of the mixer. If there is less than MAIN_TIMING_VARIANCE of samples, push an additional block here. Else, one block per GetAudioDMACallbackPeriod.

Reply:
thats smart, i'll definitely look into that